### ISSUE TITLE
[DataStore] Freeze on startup

### ISSUE THREAD

**Issue by lotchay** [aws-amplify/amplify-android#209]
Hi there, we have issue with the datastore module. Below is a snippet of the demo app:
```Java
public final class AmplifyDemoApplication extends android.app.Application {
    private static final String TAG = "AmplifyDemoApplication";

    @Override
    public void onCreate() {
        super.onCreate();
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
        try {
            AWSDataStorePlugin dataStorePlugin = AWSDataStorePlugin.forModels(AmplifyModelProvider.getInstance());
            Amplify.addPlugin(dataStorePlugin);
            Amplify.configure(getApplicationContext());
        } catch (AmplifyException exception) {
            Log.e("AmplifyGetStarted", exception.getMessage());
        }
    }
}
````
Here's my basic schema.graphql:
```graphql
type Event @model {
    id: ID!
    name: String!
    location: String
    startTime: String
    description: String
}
```
When debugging, It looks like it's stuck in an infinite loop, while trying to `Amplify.configure`

**Comment by raphkim** [comment]
Hi @lotchay 

Thanks for reporting the issue to us. I was able to reproduce this issue on my machine, and we will investigate to find out the cause.

**Comment by jamesonwilliams** [comment]
@raphkim and @lotchay Can you share your `amplifyconfiguration.json`? Which commands did you use to configured your backend, via the Amplify CLI?

**Comment by lotchay** [comment]
here's my amplifycongiuration.json:
```json
{
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "amplifyDatasource": {
                    "endpointType": "GraphQL",
                    "endpoint": "https://***.appsync-api.***.amazonaws.com/graphql",
                    "region": "***",
                    "authorizationType": "API_KEY",
                    "apiKey": "***"
                }
            }
        }
    }
}
```

**Comment by jamesonwilliams** [comment]
Thanks @lotchay; I was able to reproduce this as well. [In our `AWSDataStorePlugin`](https://github.com/aws-amplify/amplify-android/blob/master/aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java#L122-L123), we have code like:
```
someRxSingle
    .observeOn(AndroidSchedulers.mainThread())
    .blockingGet();
```

As noted in [this related comment on StackOverflow](https://stackoverflow.com/questions/56280945/use-blockingget-on-maybe-that-already-observes-on-androidschedulers-mainthread#comment99179139_56280945), if that is itself called from the main thread, it will never complete.

**This is a bug, and we should fix it in an upcoming release.**

For the time-being, a _work-around_ is to initialize Amplify from a different, non-`main` thread. There are numerous strategies to achieve this, but the most directly stated would be something like:

```
new Thread(() -> {
        try {
            AWSDataStorePlugin dataStorePlugin = AWSDataStorePlugin.forModels(AmplifyModelProvider.getInstance());
            Amplify.addPlugin(dataStorePlugin);
            Amplify.configure(getApplicationContext());
        } catch (AmplifyException exception) {
            Log.e("AmplifyGetStarted", exception.getMessage());
        }
}).start();
```
Or:
```
Executors.newCachedThreadPool().submit(() -> {
        try {
            AWSDataStorePlugin dataStorePlugin = AWSDataStorePlugin.forModels(AmplifyModelProvider.getInstance());
            Amplify.addPlugin(dataStorePlugin);
            Amplify.configure(getApplicationContext());
        } catch (AmplifyException exception) {
            Log.e("AmplifyGetStarted", exception.getMessage());
        }
});
```

**Comment by lotchay** [comment]
Thanks much for the quick work around!
Initializing Amplify using a new non main thread works. 
Happy new years =)

**Comment by jamesonwilliams** [comment]
Fixed in 0.10.0 release https://github.com/aws-amplify/amplify-android/releases/tag/release_v0.10.0
